opsdroid/core.py

```diff
@@ -303,7 +304,7 @@ def start_connectors(self, connectors):
         if connectors:
             for connector in self.connectors:
                 if self.eventloop.is_running():
-                    self.eventloop.create_task(connector.connect())
+                    asyncio.wait(self.eventloop.create_task(connector.connect()))
```

Comment:
AFAIK `asyncio.wait` is a coroutine and will return a future which never gets awaited here. What was the reasoning behind changing this?

https://docs.python.org/3/library/asyncio-task.html#asyncio.wait

Reply:
It returns an awaitable, but isn't a coroutine I don't think. The docstring says:

> Run awaitable objects in the aws set concurrently and block until the condition specified by return_when.

So by default this will block until the task finishes, which is what we desire here, so we can correctly emit the startup event knowing the connectors have all connected.

Reply:
I don't recall, but the tests pass :man_shrugging: I'll test it without as well and see if I can reverse-engineer why we did this. In the meantime @Cadair might remember better than I do.

Reply:
![image](https://user-images.githubusercontent.com/1610850/65437266-69adec80-de1b-11e9-9d90-c03ca9e45e46.png)

Reply:
![image](https://user-images.githubusercontent.com/1610850/65437367-92ce7d00-de1b-11e9-9e72-2af8dec67c99.png)

Reply:
I think you do need to await it to get it to block.

Reply:
@SolarDrew I expect everything works but there may be a warning about not awaiting it. The previous implementation also did not block at this point, it just submitted the task to the loop and moved on. 

I wondered if the idea here was to try and avoid some race condition.

Reply:
Note this is being called from inside a function which isn't `async def` so we can't await it.

The objective was to ensure that all the connectors have started before returning to the calling scope where we emit the `OpsDroidStarted` event.

Reply:
There must be a way to block in non-async code and wait for a `Task` or a `Future` to finish though?! If this isn't it, what is?

Reply:
https://docs.python.org/3/library/asyncio-task.html#asyncio.as_completed ?!

Reply:
I will have to see if I can setup a test that fails if this doesn't block...

Reply:
Check out #1072, might help.

Reply:
Ok #1072 is now merged, you'll need to deal with the conflict but you should be unblocked here now.